internal/jimmhttp/migrationproxy_handler.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2025 Canonical.
+
+package jimmhttp
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	jujuparams "github.com/juju/juju/rpc/params"
+
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/jimm"
+	"github.com/canonical/jimm/v3/internal/middleware"
+	"github.com/canonical/jimm/v3/internal/rpc"
+)
+
+// MigrationProxyHandler is an handler that provides proxying,
+// specifically for HTTP-based model migration endpoints. These
+// are used primarily to send data like charms/resources/tools
```

Comment:
It'd be good to be super explicit about what migration endpoints we're handling, even if it'd handle more. I say this because:
```go
const (
	// all endpoints managed by this handler
	migrationProxyEndpoints = "/*"
)
```

Isn't very helpful aha 😝